cdap-app-fabric/src/main/java/co/cask/cdap/gateway/handlers/ProgramLifecycleHttpHandler.java

```diff
@@ -601,14 +601,17 @@ public void stopPrograms(HttpRequest request, HttpResponder responder,
       ProgramId programId =
         Ids.namespace(namespaceId).app(program.getAppId()).program(program.getProgramType(), program.getProgramId());
       try {
-        ListenableFuture<BatchProgramResult> issuedStop = Futures.transform(lifecycleService.issueStop(programId, null),
-          new Function<ProgramController, BatchProgramResult>() {
-            @Override
-            public BatchProgramResult apply(ProgramController input) {
-              return new BatchProgramResult(program, HttpResponseStatus.OK.getCode(), null);
-            }
-          });
-        issuedStops.add(issuedStop);
+        List<ListenableFuture<ProgramController>> stops = lifecycleService.issueStop(programId, null);
+        for (ListenableFuture<ProgramController> stop : stops) {
```

Comment:
@yaojiefeng so you are adding the list of stops (multiple runs of same program id?) to the list issuedStops. but the size of the list is still the same as number of programs on L#599.  that should also be updated right?